File: 01-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dictionary = {}
        number_count = len(nums)

        for index in range(number_count):
            current_number = nums[index]
            complement_number = target - current_number

            if complement_number in dictionary:
                return [dictionary[complement_number], index]
            
            dictionary[current_number] = index
        
        return None

File: 01-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> dictionary = new Dictionary<int, int>();

        for (int index = 0; index < nums.Length; index++) {
            int currentNumber = nums[index];
            int complement = target - currentNumber;

            if (dictionary.ContainsKey(complement)) {
                int[] result = new int[] { dictionary[complement], index };
                return result;
            }

            dictionary.TryAdd(currentNumber, index);
        }

        return null;
    }
}
